src/test/java/net/minestom/server/collision/EntityBlockTouchTickIntegrationTest.java

```diff
@@ -38,37 +39,106 @@ public void onTouch(@NotNull Touch touch) {
             }
         };
 
-        instance.setBlock(0, 42, 0, Block.STONE.withHandler(handler));
-        instance.setBlock(0, 42, 1, Block.STONE.withHandler(handler));
-        instance.setBlock(0, 43, 1, Block.STONE.withHandler(handler));
-        instance.setBlock(0, 43, -1, Block.STONE.withHandler(handler));
-        instance.setBlock(1, 42, 1, Block.STONE.withHandler(handler));
-        instance.setBlock(1, 42, 0, Block.STONE.withHandler(handler));
-        instance.setBlock(0, 42, 10, Block.STONE.withHandler(handler));
+        var customBlock = Block.STONE.withHandler(handler);
+        var blockPos = Set.of(new Vec(-1, 43, 0),
+                new Vec(1, 43, 0),
+                new Vec(0, 43, -1),
+                new Vec(0, 43, 1),
+                new Vec(0, 45, 0) // Y+
+        );
 
-        var entity = new Entity(EntityType.ZOMBIE);
-        entity.setInstance(instance, new Pos(0, 42, 0.7)).join();
+        for (var pos : blockPos) {
+            instance.setBlock(pos, customBlock);
+        }
+        instance.setBlock(0, 42, 0, Block.STONE); // Regular floor as we are going to be sliding into the blocks.
 
+        var entity = new Entity(EntityType.ZOMBIE);
+        var spawnPoint = new Pos(0.5, 43, 0.5);
+        entity.setInstance(instance, spawnPoint).join();
+
+        Arrays.stream(Direction.values())
+                .filter(direction -> direction != Direction.DOWN)
+                .map(Direction::vec)
+                .forEachOrdered(vec -> {
+            entity.setVelocity(vec.mul(10));
+            entity.tick(0);
+            entity.teleport(spawnPoint);
+        });
+
+        assertEquals(blockPos, positions);
+
+        positions.clear(); // Final -Y check
+        instance.setBlock(0, 42, 0, customBlock);
         entity.tick(0);
+        assertEquals(Set.of(new Vec(0, 42, 0)), positions);
 
-        assertEquals(Set.of(new Vec(0, 42, 0),
-                new Vec(0, 42, 1),
-                new Vec(0, 43, 1)),
-                positions);
+        assertEquals(instance, entity.getInstance());
+    }
+    @Test
+    public void entityPhysicsCheckTouchTickFarPositiveXNegativeZ(Env env) {
```

Comment:
Nitpick: Remove public for junit5.
Ref: [junit.org/junit5/docs/snapshot/user-guide#writing-tests-classes-and-methods](https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-classes-and-methods)